components/bookshelf/Book.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react"
+import Image from "next/image"
+import {
+  BookCard,
+  BookContent,
+  BookImage,
+  HeaderBar,
+  InfoBox,
+  Author,
+  Publisher,
+  Title,
+  Synopsis,
+  SideBar,
+  ExtendedSideBar,
+} from "../styled/bookshelf"
+import { BookInterface } from "../../api/types"
+
+interface BookProps {
+  book: BookInterface
+}
+
+const Book = ({ book }: BookProps) => {
+  const [isAdding, setIsAdding] = useState<boolean>(false)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+
+  const addToList = () => {
+    setIsAdding(true)
+    setIsLoading(true)
+    setTimeout(() => {
+      setIsLoading(false)
+    }, 2500)
```

Comment:
I'm assuming this will be covnerted to an API call in another PR?

Reply:
yep!